File: backend/schemas/medical_record.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import date, datetime


class MedicalRecordBase(BaseModel):
    record_type: str  # vaccination, checkup, surgery, illness, injury, medication, other
    title: str
    description: Optional[str] = None
    date: date
    veterinarian: Optional[str] = None
    clinic: Optional[str] = None
    follow_up_date: Optional[date] = None
    attachments: Optional[str] = None  # JSON string of file URLs
    notes: Optional[str] = None
    is_completed: bool = True


class MedicalRecordCreate(MedicalRecordBase):
    pet_id: int


class MedicalRecordUpdate(MedicalRecordBase):
    pass


class MedicalRecordRead(MedicalRecordBase):
    id: int
    pet_id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


# Response models for API
class MedicalRecordListResponse(BaseModel):
    records: List[MedicalRecordRead]
    total: int
    page: int
    page_size: int


class MedicalRecordSummary(BaseModel):
    pet_id: int
    total_records: int
    recent_checkup: Optional[date] = None
    next_followup: Optional[date] = None
    vaccination_count: int
    surgery_count: int

File: backup_20250923_180339/frontend/src/types/medical/medicalRecord.ts
// Medical Record Types for PawfectPal

export type MedicalRecordType = 
  | 'vaccination' 
  | 'checkup' 
  | 'surgery' 
  | 'illness' 
  | 'injury' 
  | 'medication' 
  | 'other';

export interface MedicalRecord {
  id?: number;
  petId: number;
  recordType: MedicalRecordType;
  title: string;
  description?: string;
  date: string; // ISO date string
  veterinarian?: string;
  clinic?: string;
  followUpDate?: string; // ISO date string
  attachments?: string[]; // URLs to documents/images
  notes?: string;
  isCompleted: boolean;
  createdAt?: string; // ISO datetime string
  updatedAt?: string; // ISO datetime string
}

export interface MedicalRecordCreate {
  petId: number;
  recordType: MedicalRecordType;
  title: string;
  description?: string;
  date: string;
  veterinarian?: string;
  clinic?: string;
  followUpDate?: string;
  attachments?: string;
  notes?: string;
  isCompleted?: boolean;
}

export interface MedicalRecordUpdate {
  recordType: MedicalRecordType;
  title: string;
  description?: string;
  date: string;
  veterinarian?: string;
  clinic?: string;
  followUpDate?: string;
  attachments?: string;
  notes?: string;
  isCompleted: boolean;
}

export interface MedicalRecordListResponse {
  records: MedicalRecord[];
  total: number;
  page: number;
  pageSize: number;
}

export interface MedicalRecordSummary {
  petId: number;
  totalRecords: number;
  recentCheckup?: string; // ISO date string
  nextFollowup?: string; // ISO date string
  vaccinationCount: number;
  surgeryCount: number;
}

export interface MedicalRecordFilters {
  recordType?: MedicalRecordType;
  dateFrom?: string;
  dateTo?: string;
  veterinarian?: string;
  clinic?: string;
  isCompleted?: boolean;
  page?: number;
  pageSize?: number;
}

